trieste/space.py

```diff
@@ -241,11 +247,48 @@ def __contains__(self, value: TensorType) -> bool | TensorType:
         return tf.reduce_all(value >= self._lower) and tf.reduce_all(value <= self._upper)
 
     def sample(self, num_samples: int) -> TensorType:
+        """
+        Sampling points randomly
+
+        :param num_samples: The number of points to sample from this search space.
+        :return: ``num_samples`` i.i.d. random points, sampled uniformly, and without replacement,
+            from this search space.
+        """
         dim = tf.shape(self._lower)[-1]
         return tf.random.uniform(
             (num_samples, dim), minval=self._lower, maxval=self._upper, dtype=self._lower.dtype
         )
 
+    def sample_halton(self, num_samples: int, seed: int = 0) -> TensorType:
+        """
+        Sampling points using randomized halton sequence
+
+        :param num_samples: The number of points to sample from this search space.
+        :param seed: Random seed for the halton sequence
+        :return: ``num_samples`` of points, using halton sequence.
+        """
+        if num_samples == 0:
+            return []
```

Comment:
why do we have this behaviour?

Reply:
we can use 0 as the num_samples in tf.random.uniform it will return empty tensor. this is not the case for Sobol and Halton, because using 0 num_samples will raise an error. to make things have the same behaviors I add this line.

Reply:
Okay. Please can you write a test to check that this behaviour happens?

Reply:
will do!

Reply:
wait, actually, this test is included in the `test_box_sampling_returns_correct_shape` we have 0 as test param there.